### ISSUE TITLE
strike() doesn't apply to single characters in math equation

### ISSUE THREAD

**Issue by Leedehai** [typst/typst#1263]
### Description

In a math expression, `#strike()` is unable to put a horizontal strike on a single character.

```
$ #strike("l") #strike("la") $
```
<img width="104" alt="Screenshot 2023-05-20 at 00 33 54" src="https://github.com/typst/typst/assets/18319900/f938ddb6-7a02-41c2-936e-61d3b611e577">

(indeed, I can use [`cancel()`](https://typst.app/docs/reference/math/cancel/) to strike out content, but it doesn't offer horizontal lines, and it's difficult to get the custom angle right -- rotating by 45deg isn't guaranteed to yield a horizontal line according to cancel.rs)

### Operating system

macOS

### Typst version

- [X] I am using the latest version of Typst

**Comment by laurmaedje** [comment]
Yeah, `strike` is not designed for math mode and happens to work for two letters because they go through normal text layout. Medium-term, we need to find a good way to integrate math and normal layout more closely, there are lots more examples where they don't interact well.

**Comment by Leedehai** [comment]
Thanks. Maybe it's a good idea to create a label (e.g. "normal-vs-math-layout") for this family of GitHub issues? Your decision, of course :)

**Comment by Enivex** [comment]
Note that `cancel` can now be used for this purpose due to https://github.com/typst/typst/pull/2466, though `strike` still does not work.

**Comment by Ants-Aare** [comment]
@laurmaedje 
Also strike doesn't seem to like when it has math children:

`#strike[Hello #math.attach("H", br:"2", tr: "2+") world]`

<img width="256" alt="Image" src="https://github.com/user-attachments/assets/87370570-2c6d-4576-96e6-c2dbdabbdaab" />

I personally would expect the line to go through the H at the same height as the hello world, but instead it strikes the each of the children. But apparently only if the children are not single characters: 2+ at the top, but doesn't strike the 2 at the bottom

**Comment by mkorje** [comment]
For single character strings in math this has been fixed (see https://github.com/typst/typst/pull/5738). For other math content this is a duplicate of #2200.
